### PR TITLE
Add a very high-level overview of CTCP in guides

### PR DESCRIPTION
**Pull request by astorije**
Extracted from #84.

### CODE REVIEW

File: _guides/ctcp.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  `VERSION thelounge v3.0.0 -- https://thelounge.chat/`
+- `/ctcp <nick> source` returns
+  `SOURCE git+https://github.com/thelounge/thelounge.git`
+- `/ctcp <nick> ping foo` returns `PING foo`
+
+To know more about internal details of how The Lounge implements CTCP responses,
+take a look at
+[the related source code](https://github.com/thelounge/thelounge/blob/master/src/plugins/irc-events/ctcp.js).
+
+The Lounge does not support all CTCP command. In particular, it does not
```
Comment (xPaw) [inline _guides/ctcp.md]: `CTCP commands`

File: _guides/ctcp.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+currently support
+[DCC (Direct Client-to-Client)](https://en.wikipedia.org/wiki/Direct_Client-to-Client).
+
+A note of warning: when querying other users, know that their clients might
+inform them that you are requesting information about them, in the form of a
+notification or sound.
+
+If you want to learn more about CTCP, [this document](http://modern.ircdocs.horse/ctcp.html) is a great reference, more
```
Comment (xPaw) [inline _guides/ctcp.md]: That page links to a RFC (draft) now: https://tools.ietf.org/html/draft-oakley-irc-ctcp-02

Comment (astorije) [reply _guides/ctcp.md]: Good catch. I think https://rawgit.com/DanielOaks/irc-rfcs/master/dist/draft-oakley-irc-ctcp-latest.html is much more reader-friendly and effectively has the same content. Are you okay if I use this one instead?

File: _guides/ctcp.md (hunk 1 above)
Comment (astorije) [reply _guides/ctcp.md]: Good catch, fixing.

File: _guides/ctcp.md (hunk 2 above)
Comment (astorije) [reply _guides/ctcp.md]: > Rawgit will die soon.

What do you mean?

Comment (xPaw) [reply _guides/ctcp.md]: See https://rawgit.com/

Comment (astorije) [reply _guides/ctcp.md]: Ah, right, I didn't catch that. Switching to the IETF link now.

**Pull request merged**